gdx/test/com/badlogic/gdx/utils/LongArrayTest.java

```diff
@@ -0,0 +1,425 @@
+		}
+	}
+
+	/** Test of the setSize() method */
+	@Test
+	public void setSizeTest () {
+		LongArray longArray1 = new LongArray(
+			new long[] {1, 2, 4, 6, 32, 53, 564, 53, 2, 1, 89, 90, 10, 389, 8, 392, 4, 27346, 2, 234, 12});
+		longArray1.setSize(23);
+		Assert.assertEquals(23, longArray1.size);
+		longArray1.setSize(10);
+		Assert.assertArrayEquals(new long[] {1, 2, 4, 6, 32, 53, 564, 53, 2, 1}, longArray1.toArray());
+		LongArray longArray2 = new LongArray(new long[] {1, 2, 3});
+		Assert.assertArrayEquals(new long[] {1, 2, 3, 0, 0, 0, 0, 0}, longArray2.setSize(5));
+		try {
+			longArray1.setSize(-3);
+			Assert.fail();
+		} catch (IllegalArgumentException e) {
+			// This should throw an exception as it is not allowed to have a negative integer as setSize argument
+		}
+	}
+
+	/** Test of resize() method */
+	@Test
+	public void resizeTest () {
+		LongArray longArray1 = new LongArray(
+			new long[] {1, 2, 4, 6, 32, 53, 564, 53, 2, 1, 89, 90, 10, 389, 8, 392, 4, 27346, 2, 234, 12});
+		Assert.assertArrayEquals(
+			new long[] {1, 2, 4, 6, 32, 53, 564, 53, 2, 1, 89, 90, 10, 389, 8, 392, 4, 27346, 2, 234, 12, 0, 0},
+			longArray1.resize(23));
+	}
+
+	/** Test of sort() and reverse() methods */
+	@Test
+	public void sortAndReverseTest () {
+		LongArray longArray1 = new LongArray(
+			new long[] {1, 2, 4, 6, 32, 53, 564, 53, 2, 1, 89, 90, 10, 389, 8, 392, 4, 27346, 2, 234, 12});
+		longArray1.sort();
+		Assert.assertArrayEquals(new long[] {1, 1, 2, 2, 2, 4, 4, 6, 8, 10, 12, 32, 53, 53, 89, 90, 234, 389, 392, 564, 27346},
+			longArray1.toArray());
+		longArray1.reverse();
+		Assert.assertArrayEquals(new long[] {27346, 564, 392, 389, 234, 90, 89, 53, 53, 32, 12, 10, 8, 6, 4, 4, 2, 2, 2, 1, 1},
+			longArray1.toArray());
+	}
+
+	/** Test of equals() method */
+	@Test
+	public void equalsTest () {
+		LongArray longArray1 = new LongArray();
+		LongArray longArray2 = new LongArray();
+		longArray1.add(1, 2);
+		longArray2.add(1, 2);
+
+		Assert.assertTrue(longArray1.equals(longArray2));
+
+		// Verifying that an object of instance =/= LongArray cannot be equal
+		ArrayMap<Integer, Integer> o = new ArrayMap<Integer, Integer>(); // Random object of a different class
+		Assert.assertFalse(longArray1.equals(o));
+
+		// An unordered array, even if the content is the same, cannot be equal to an ordered array
+		LongArray longArray3 = new LongArray(false, 16);
+		longArray3.add(1, 2);
+		Assert.assertFalse(longArray1.equals(longArray3));
+
+		// The capacity isn't something that makes two arrays not equal
+		LongArray longArray4 = new LongArray(true, 12);
+		longArray4.add(1, 2);
+		Assert.assertTrue(longArray1.equals(longArray4));
+
+		// Standard verification that two arrays with different content are not equal
+		longArray1.add(3);
+		Assert.assertFalse(longArray1.equals(longArray2));
+	}
+
+	/** Test of the toString() methods */
+	@Test
+	public void toStringTest () {
```

Comment:
I'm also reiterating Frosty-J's concern about the toString format being flexible to change; we might want to switch to `{}` instead of `[]` to work as an array literal, and append `L` to each number so this could go into source code. I don't think that's terribly likely, but it shouldn't be codified in a test.